R/mzQCjson.R

```diff
@@ -0,0 +1,255 @@
+#' 
+#' 
+#' 
+#' @param qcMetric A list containing all metric information
+#' @param data data frame of all cv Parameter
+#' @return data frame of all metric that appears in PTXQC
+#' 
+#' @export
+#' 
+#' 
+#' 
+readvalue <-  function(qcMetric, data){
+  runQuality = FALSE
+  setQuality = FALSE
+  new_metric = length(data$cvRef) + 1
+  for(klass in qcMetric){
+    index <- 1
+    for(metric_name in klass$qcCV){
+      ##If this metric exists in data, other information about this metric is extracted from data, 
+      ##or if not, a new line of data is added to data and a warning is given
+      if(metric_name %in% data$name) {
+        if(! is.na(data$value[which(metric_name == data$name)])) warning("Repeat metric! Each metric has only one. Please compare the data entered before with the existing data and Retain original data.")
+        else {
+          locate = which(data$name == metric_name)
+          data$value[locate] = rjson::toJSON(klass$mzQCdata[index])
+          data$quality_type[locate] = klass$quality_type[index]
+          data$input_file[locate] = klass$input_file[index]
+          
+        }
+      }else{
+        warning("There is no this metric in cv Term of mzQC, please write to the mailing list psidev-qc-dev@lists.sourceforge.net to update mzQC-metrics list.")
+        name = metric_name
+        value =  rjson::toJSON(klass$mzQCdata[index])
+        quality_type = klass$quality_type[index]
+        input_file = klass$input_file[index]
+        data[new_metric, ] <- c(NA, NA, metric_name, NA, value, quality_type, input_file)
+        new_metric = new_metric + 1
+      }
+      index = index + 1
+    }
+  }
+  ##Check whether "..Ecoli_01" or "..Ecoli_02" is included in metric's value
+  for(klass in qcMetric){                                                                                                                 index <- 1
+    for(mzQC in klass$mzQCdata){
+      if("fc.raw.file" %in% colnames(mzQC)){
+        if("..Ecoli_01" %in% mzQC$fc.raw.file || "..Ecoli_02" %in% mzQC$fc.raw.file){  
+          if(unlist(klass$quality_type[index] == "runQuality")) runQuality = TRUE
+          else setQuality = TRUE                                                                                                          }
+      }
+    index = index + 1
+  }                                                                                                                               }
+  
+  ##delete the row where the value of data is NA 
+  data <- data[! is.na(data$value), ]
+  if(runQuality == TRUE) data[length(data$quality_type) + 1, ] <- c(NA, NA, NA, NA, NA, "runQuality", "Raw file: Ecoli")
+  if(setQuality == TRUE) data[length(data$quality_type) + 1, ] <- c(NA, NA, NA, NA, NA, "setQuality", "Raw file: Ecoli")
+  
+  return(data)
+}
+
+#'
+#' Extract the data of each row in data and store it in a list
+#' 
+#' 
+#' 
+#' @param data A list 
+#' @return a list
+#' 
+#' @export
+#' 
+#' 
+#' 
+transform_data <-function(data){
```

Comment:
have you tried using `tr()`?

Reply:
you mean transpose data.frame? I've only tried `t()` before

Reply:
thats what I meant. Sorry.
If that did not work, ok. Then leave that as is for the time being